crates/miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -735,6 +739,68 @@ export.input_note_get_assets_info
     # => [ASSETS_COMMITMENT, num_assets, pad(11)]
 end
 
+#! Returns the recipient of the input note with the specified index.
+#!
+#! Inputs:  [note_index, pad(15)]
+#! Outputs: [RECIPIENT, pad(12)]
+#!
+#! Where:
+#! - note_index is the index of the input note whose recipient should be returned.
+#! - RECIPIENT is the commitment to the input note's script, inputs, the serial number.
+#!
+#! Panics if:
+#! - the note index is greater or equal to the total number of input notes.
+#!
+#! Invocation: dynexec
+export.input_note_get_recipient
+    # assert that the provided note index is less than the total number of input notes
+    exec.input_note::assert_note_index_in_bounds
+    # => [note_index, pad(15)]
+
+    # get the note data pointer by the provided index
+    exec.memory::get_input_note_ptr
+    # => [note_ptr, pad(15)]
+
+    # get the recipient
+    exec.memory::get_input_note_recipient
+    # => [RECIPIENT, pad(15)]
+
+    # truncate the stack
+    swapw dropw
+    # => [RECIPIENT, pad(12)]
```

Comment:
Assuming the previous state is `[RECIPIENT, pad(15)]`, wouldn't this result in `[RECIPIENT, pad(11)]`? i.e., it would drop an extra element from the stack.

Reply:
Indeed. The idea was to save a bit of cycles, but since the fast processor handles underflow differently, it makes sense to drop only actual overflowing elements.

Reply:
I also think that because these procedures are invoked via `dynexec`, we shouldn't assume that the rest of the stack has useless info. Not an issue in this specific case because we call the procedures from the same file, but it is still cleaner not to assume which context the procedure will be called from.